src/query_plan/fetch_dependency_graph.rs

```diff
@@ -2429,13 +2422,56 @@ fn compute_input_rewrites_on_key_fetch(
     }
 }
 
+/// Returns an updated pair of (`operation`, `defer_context`) after the `defer` directive removed.
+/// - The updated operation can be `None`, if operation is no longer necessary.
 fn extract_defer_from_operation(
-    _dependency_graph: &mut FetchDependencyGraph,
-    _operation: &OpPathElement,
-    _defer_context: &DeferContext,
-    _node_path: &FetchDependencyGraphNodePath,
-) -> (Option<OpPathElement>, DeferContext) {
-    todo!() // Port `extractDeferFromOperation`
+    dependency_graph: &mut FetchDependencyGraph,
+    operation: &OpPathElement,
+    defer_context: &DeferContext,
+    node_path: &FetchDependencyGraphNodePath,
+) -> Result<(Option<OpPathElement>, DeferContext), FederationError> {
+    let defer_args = operation.defer_directive_args();
+    let Some(defer_args) = defer_args else {
+        let updated_path_to_defer_parent = defer_context
+            .path_to_defer_parent
+            .with_pushed(operation.clone().into());
+        let updated_context = DeferContext {
+            path_to_defer_parent: updated_path_to_defer_parent.into(),
+            // Following fields are identical to those of `defer_context`.
+            current_defer_ref: defer_context.current_defer_ref.clone(),
+            active_defer_ref: defer_context.active_defer_ref.clone(),
+            is_part_of_query: defer_context.is_part_of_query,
+        };
+        return Ok((Some(operation.clone()), updated_context));
+    };
+
+    let Some(ref updated_defer_ref) = defer_args.label else {
```

Comment:
Use the existing `label` method:
```suggestion
    let Some(updated_defer_ref) = defer_args.label() else {
```